### PR TITLE
topological sorting for building onnx model

### PR DESCRIPTION
**Pull request by jiafatom**

### CODE REVIEW

File: onnxconverter_common/optimizer.py [hunk 1]
```diff
@@ -723,6 +724,73 @@ def _build_onnx_model(node_list):
     return regenerated
 
 
+def _visit(name_to_node_map, n_name, result):
+    node = name_to_node_map[n_name]
+    if node.status == 'perm':
```
Comment (wenbingl) [inline onnxconverter_common/optimizer.py]: a boolean variable seems enough for this case, None, False, True.

File: onnxconverter_common/optimizer.py [hunk 2]
```diff
@@ -723,6 +724,73 @@ def _build_onnx_model(node_list):
...
+
+
+def _topological_sort(node_list):
+    result = []
+    name_set = set()
+    name_idx = 0
+    name_to_node_map = dict()
+    prefix = "abc_"
+
+    node_list_mark = copy.deepcopy(node_list)
```
Comment (wenbingl) [inline onnxconverter_common/optimizer.py]: why do you clone this node list?

File: onnxconverter_common/optimizer.py (hunk 1 above)
Comment (jiafatom) [reply onnxconverter_common/optimizer.py]: Using string can be more explicitly read.

File: onnxconverter_common/optimizer.py [hunk 3]
```diff
@@ -723,6 +724,73 @@ def _build_onnx_model(node_list):
...
+
+
+def _get_unmark_node(name_to_node_map):
+    for k, v in six.iteritems(name_to_node_map):
+        if v.status == 'unmark':
+            return k
+    return None
+
+
+def _topological_sort(node_list):
```
Comment (wenbingl) [inline onnxconverter_common/optimizer.py]: The order is input_node at the first place of the node list?

Comment (jiafatom) [reply onnxconverter_common/optimizer.py]: yes

File: onnxconverter_common/optimizer.py (hunk 2 above)
Comment (jiafatom) [reply onnxconverter_common/optimizer.py]: ok, we can modify the original node_list property, will avoid copy.

**Pull request merged**